Simplenote/src/main/java/com/automattic/simplenote/viewmodels/AddTagViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.automattic.simplenote.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.automattic.simplenote.R
+import com.automattic.simplenote.models.Tag
+import com.automattic.simplenote.utils.TagUtils
+import com.simperium.client.Bucket
+import com.simperium.client.BucketObjectNameInvalid
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+
+class AddTagViewModel(
+        private val tagsBucket: Bucket<Tag>,
+        defaultDispatcher: CoroutineDispatcher = Dispatchers.Default
+) : ScopedViewModel(defaultDispatcher) {
+
+    private val _showKeyboard = MutableLiveData<Boolean>()
+    val showKeyboard: LiveData<Boolean> = _showKeyboard
+
+    private val _tagError = MutableLiveData<Int?>()
+    val tagError: LiveData<Int?> = _tagError
```

Comment:
I'd intuitively replace this by an `UiState` object which would contain both the error and the text value. Both would be populated in the `validateTag` call and the `text` value would be used on save instead of retrieving the value from the field. Let me know what you think! Maybe it's out of scope of this PR. This UI State could contain the `showKeyboard` as well.

Reply:
I have added a `UiState` field and also one to handle the events. The only concern with the current solution is that it creates a new UiState for each text change in the field. 

Fixed in ff0380dcc28559b1f4652e58177aa23988a8e8e0.